### PR TITLE
server variable

### PR DESCRIPTION
**Pull request by yunheMsft**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json [hunk 1]
```diff
@@ -3098,13 +3193,6 @@
           "items": {
             "type": "string"
           }
-        },
-        "transforms": {
-          "description": "List of transforms",
-          "type": "array",
-          "items": {
-            "$ref": "#/definitions/transform"
-          }
```
Comment (yunheMsft) [inline specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json:3107]: Should we remove transform

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json [hunk 2]
```diff
@@ -3756,6 +3844,89 @@
...
+          "type": "string",
+          "enum": [
+            "DeliveryRuleSocketAddressConditionParameters"
+          ]
+        },
+        "operator": {
+          "description": "Describes operator to be matched",
+          "type": "string",
+          "enum": [
+            "IPMatch"
```
Comment (huaiyizhu) [inline specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json:3868]: from PM spec, we have IPMatch and GeoMatch. Please confirm with @jessie.

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json [hunk 3]
```diff
@@ -3756,6 +3844,89 @@
...
+            "Any",
+            "Equal",
+            "Contains",
+            "BeginsWith",
+            "EndsWith",
+            "LessThan",
+            "LessThanOrEqual",
+            "GreaterThan",
+            "GreaterThanOrEqual",
+            "RegEx"
```
Comment (huaiyizhu) [inline specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json:3914]: RegEx operator seems not available for ClientPort/ServerPort, please confirm with @jessie

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json [hunk 4]
```diff
@@ -3766,6 +3937,143 @@
...
+          "type": "string",
+          "enum": [
+            "DeliveryRuleServerPortConditionParameters"
+          ]
+        },
+        "operator": {
+          "description": "Describes operator to be matched",
+          "type": "string",
+          "enum": [
+            "Any",
```
Comment (huaiyizhu) [inline specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json:3951]: I suppose operator of ServerPort and ClientPort should be align, let's confirm with @jessie.

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json [hunk 5]
```diff
@@ -3766,6 +3937,143 @@
...
+          "type": "boolean"
+        },
+        "matchValues": {
+          "description": "The match value for the condition of the delivery rule",
+          "type": "array",
+          "items": {
+            "type": "string",
+		        "enum": [
+              "PCTv1",
+              "SSLv2",
```
Comment (huaiyizhu) [inline specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json]: TLSv1, TLSv1.1, TLSv1.2 as selected

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json [hunk 6]
```diff
@@ -2801,7 +2801,12 @@
             "UrlFileName",
             "HttpVersion",
             "Cookies",
-            "IsDevice"
+            "IsDevice",
+            "SocketAddress",
+            "ClientPort",
+            "ServerPort",
+            "Hostname",
```
Comment (1VarunCh1) [inline specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json]: Should we use HostName to be consistent?

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json (hunk 4 above)
Comment (yunheMsft) [reply specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json:3951]: Updated

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json (hunk 2 above)
Comment (yunheMsft) [reply specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json:3868]: Added GeoMatch

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json (hunk 3 above)
Comment (yunheMsft) [reply specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json:3914]: updated

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json (hunk 4 above)
Comment (yunheMsft) [reply specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json:3951]: updated according to the doc

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json (hunk 6 above)
Comment (yunheMsft) [reply specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json]: updated casing

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json (hunk 5 above)
Comment (yunheMsft) [reply specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json]: updated

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json (hunk 3 above)
Comment (yunheMsft) [reply specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json:3914]: removed

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json [hunk 7]
```diff
@@ -2801,7 +2801,12 @@
             "UrlFileName",
             "HttpVersion",
             "Cookies",
-            "IsDevice"
+            "IsDevice",
+            "SocketAddress",
```
Comment (aramesh-msft) [inline specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json]:
```suggestion
            "SocketAddr",
```

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json [hunk 8]
```diff
@@ -2801,7 +2801,12 @@
...
+            "ClientPort",
+            "ServerPort",
+            "HostName",
+            "SSLProtocol"
```
Comment (aramesh-msft) [inline specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json]:
```suggestion
            "SslProtocol"
```

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json [hunk 9]
```diff
@@ -3756,6 +3844,173 @@
...
+            "DeliveryRuleSocketAddressConditionParameters"
+          ]
+        },
+        "operator": {
+          "description": "Describes operator to be matched",
+          "type": "string",
+          "enum": [
+            "Any",
+            "IPMatch",
+            "GeoMatch"
```
Comment (aramesh-msft) [inline specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json]: remove geomatch. See MatchConditionValidator.cs in shared-protection for details.

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json [hunk 10]
```diff
@@ -3756,6 +3844,173 @@
...
+        "operator": {
+          "description": "Describes operator to be matched",
+          "type": "string",
+          "enum": [
+            "Any",
+            "Equal",
+            "Contains",
+            "BeginsWith",
+            "EndsWith",
+            "RegEx"
```
Comment (aramesh-msft) [inline specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json:4006]: Add numeric operators also just like any other condition. The numeric operator will check length of string.

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json [hunk 11]
```diff
@@ -3756,6 +3844,173 @@
...
+          "description": "Describes operator to be matched",
+          "type": "string",
+          "enum": [
+            "Any",
+            "Equal",
+            "Contains",
+            "LessThan",
+            "LessThanOrEqual",
+            "GreaterThan",
+            "GreaterThanOrEqual"
```
Comment (aramesh-msft) [inline specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json]: Add string operators also just like any other condition. The clientport and serverport are actually interpreted as strings in DP.

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json [hunk 12]
```diff
@@ -3756,6 +3844,173 @@
...
```
Comment (aramesh-msft) [inline specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json]: Add string operators also just like any other condition. The clientport and serverport are actually interpreted as strings in DP.

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json (hunk 7 above)
Comment (aramesh-msft) [reply specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json]: Possibly change everywhere it shows

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json (hunk 8 above)
Comment (aramesh-msft) [reply specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json]: Possibly change everywhere it shows

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json (hunk 7 above)
Comment (yunheMsft) [reply specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json]: updated

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json (hunk 8 above)
Comment (yunheMsft) [reply specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json]: updated

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json (hunk 9 above)
Comment (yunheMsft) [reply specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json]: removed

File: specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json (hunk 10 above)
Comment (yunheMsft) [reply specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/cdn.json:4006]: Added numeric operators back

**Review by jorinmejia — APPROVED** [review]
LGTM

**Pull request merged**